nixos/modules/services/misc/tiddlywiki.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.tiddlywiki;
+  listenParams = concatStrings (mapAttrsToList (n: v: " '${n}=${toString v}' ") cfg.listen);
+  exe = "${cfg.package}/lib/node_modules/.bin/tiddlywiki";
+
+in {
+
+  options.services.tiddlywiki = {
+
+    enable = mkEnableOption "TiddlyWiki nodejs server";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.nodePackages.tiddlywiki;
+      defaultText = "pkgs.nodePackages.tiddlywiki";
+      description = "TiddlyWiki package to use.";
+    };
+
+    wikipath = mkOption {
+      type = types.path;
+      default = "/var/lib/tiddlywiki";
+      description = "Path to wiki folder.";
+    };
+
+    listen = mkOption {
```

Comment:
The option `listen` is mostly used by other modules to specify the bind address. It seems like `listenOptions` or `extraOptions` might make more sense.

Reply:
that makes sense, renamed to `listenOptions`